src/lib/storage/mvcc_data.hpp

```diff
@@ -37,10 +44,19 @@ struct MvccData {
   void shrink();
 
   /**
-   * Grows mvcc data by the given delta
+   * Grows mvcc data by the given delta, guarded by the table's append_mutex.
```

Comment:
For me that sounds like `grow_by` is somehow ensuring that the table's mutex is guarding it. But the mutex is only guarding the operation because the Insert operator is acquiring it, right?

Reply:
"The caller should guard this using the table's append_mutex." - better?

Reply:
Much better!